esmvalcore/_recipe.py

```diff
@@ -990,13 +990,13 @@ def _get_preprocessor_task(variables, profiles, config_user, task_name):
                     _get_correct_fx_file(var, fx_var, config_user)[1]
                     for fx_var in fx_vars
                 ]
-                # list may be (intentionally) empty - catch it here
-                if not fx_variables:
-                    raise RecipeError(
-                        f"Preprocessor task for {fx_var} of {var}"
-                        f"can not be performed since there is "
-                        f"no data available. ")
                 for fx_variable in fx_variables:
+                    # list may be (intentionally) empty - catch it here
+                    if not fx_variable:
+                        raise RecipeError(
+                            f"One or more of {fx_vars} are missing data. "
```

Comment:
This is still not a useful error message. can we change it to:

```suggestion
                            f"One or more of {step} fx data for of {var['short_name']} are missing. "
```